code/_onclick/hud/spell_screen_objects.dm

```diff
@@ -144,7 +144,7 @@
 	..()
 	if(spellmaster)
 		spellmaster.spell_objects -= src
-	if(spellmaster && !spellmaster.spell_objects.len)
+	if(spellmaster && !spellmaster.spell_objects.len && master)
```

Comment:
What even is master in this instance?
I think that should be returnToPool(spellmaster)